packages/api/src/codegen/languages/typescript/index.ts

```diff
@@ -137,49 +119,62 @@ export default class TSGenerator extends CodeGenerator {
-
     this.types = new Map();
     this.schemas = {};
   }
 
+  // eslint-disable-next-line class-methods-use-this
   async install(storage: Storage, opts: InstallerOptions = {}): Promise<void> {
     const installDir = storage.getIdentifierStorageDir();
 
-    const info = this.spec.getDefinition().info;
-    let pkgVersion = semver.coerce(info.version);
-    if (!pkgVersion) {
-      // If the version that's in `info.version` isn't compatible with semver NPM won't be able to
-      // handle it properly so we need to fallback to something it can.
-      pkgVersion = semver.coerce('0.0.0') as SemVer;
-    }
+    const npmInstall = ['install', '--save', opts.dryRun ? '--dry-run' : ''].filter(Boolean);
 
-    const pkg: PackageJson = {
-      name: `@api/${storage.identifier}`,
-      version: pkgVersion.version,
-      main: `./index.${this.outputJS ? 'js' : 'ts'}`,
-      types: './index.d.ts', // Types are always present regardless if you're getting compiled JS.
-    };
+    // This will install the installed SDK as a dependency within the current working directory,
+    // adding `@api/<sdk identifier>` as a dependency there so you can load it with
+    // `require('@api/<sdk identifier>)`.
+    await execa('npm', [...npmInstall, installDir].filter(Boolean))
+      .then(res => {
+        if (opts.dryRun) {
+          (opts.logger ? opts.logger : logger)(res.command);
+          (opts.logger ? opts.logger : logger)(res.stdout);
+        }
+      })
+      .catch(err => {
+        // If `npm install` throws this error it always happens **after** our dependencies have been
+        // installed and is an annoying quirk that sometimes occurs when installing a package within
+        // our workspace as we're creating a circular dependency on `@readme/api-core`.
+        if (
+          process.env.NODE_ENV === 'test' &&
+          err.message.includes("npm ERR! Cannot set properties of null (setting 'dev')")
+        ) {
+          (opts.logger ? opts.logger : logger)("npm threw an error but we're ignoring it");
+          return;
+        }
 
-    fs.writeFileSync(path.join(installDir, 'package.json'), JSON.stringify(pkg, null, 2));
+        if (opts.dryRun) {
+          (opts.logger ? opts.logger : logger)(err.message);
+          return;
+        }
 
-    const npmInstall = ['install', '--save', opts.dryRun ? '--dry-run' : ''].filter(Boolean);
+        throw err;
+      });
+  }
 
-    // This will install packages required for the SDK within its installed directory in `.apis/`.
-    await execa('npm', [...npmInstall, ...Object.keys(this.requiredPackages)].filter(Boolean), {
+  /**
+   * Compile the TS code we generated into JS for use in CJS and ESM environments.
+   *
+   */
+  // eslint-disable-next-line class-methods-use-this
+  async compile(storage: Storage, opts: InstallerOptions = {}): Promise<void> {
+    const installDir = storage.getIdentifierStorageDir();
+
+    await execa('npm', ['install', 'tsup', 'typescript', '-D'], {
```

Comment:
per this comment, might be good to remove this and take care of the dev-dep installation in the earlier install command so we're not running `npm i` several times